app/src/pages/eligibility.tsx

```diff
@@ -1,76 +1,122 @@
-    labelKey: 'continue',
-    route: incomeRoute,
-  })
-  const [form, setForm] = useState(session?.eligibility)
-  const requiredMet = () => {
-    const categorical = [
-      'pregnant',
-      'baby',
-      'child',
-      'guardian',
-      'pregnant',
-      'none',
-    ].some((category) => form[category as keyof typeof form])
-    const programs = ['insurance', 'snap', 'tanf', 'none2'].some(
-      (program) => form[program as keyof typeof form]
+  // Function to check whether all the required fields in this form
+  // page have been filled out.
+  // @TODO: This could be further refactored to be more generic.
+  const isRequiredMet = (formToCheck: EligibilityData) => {
+    return (
+      formToCheck.residential !== '' &&
+      formToCheck.categorical.length > 0 &&
+      formToCheck.previouslyEnrolled !== '' &&
+      formToCheck.adjunctive.length > 0
     )
+  }
 
-    return form.residential && categorical && form.before && programs
+  // Function to update button route.
+  // Route the user to /other-benefits if they do not meet basic eligibility requirements.
+  // Route the user to /income if they do not meet adjunctive eligibility requirements.
+  // Otherwise, route the user to /choose-clinic.
+  const getRouting = (formToCheck: EligibilityData) => {
+    const isResident = formToCheck.residential === 'yes'
+    const hasCategory = !formToCheck.categorical.includes('none')
+    if (!isResident || !hasCategory) {
+      return '/other-benefits'
+    } else if (formToCheck.adjunctive.includes('none')) {
+      return '/income'
+    } else {
+      return '/choose-clinic'
+    }
   }
-  const [disabled, setDisabled] = useState<boolean>(!requiredMet())
 
-  useEffect(() => {
-    /* NOTE: We are using useEffect() because we want to make sure the props provided by getServerSideProps() are reliably loaded into the page. */
-    const prevRouteIndex = props.previousRoute.lastIndexOf('/')
-    const previousRoute = props.previousRoute.substring(prevRouteIndex)
-    if (form.none) {
-      setContinueBtn({ ...continueBtn, route: '/other-benefits' })
-    } else if (previousRoute === '/review') {
-      setContinueBtn({
-        labelKey: 'updateAndReturn',
-        route: previousRoute,
-      })
-    } else setContinueBtn({ ...continueBtn, route: incomeRoute })
-  }, [form.none, props.previousRoute])
+  // Set up action button and routing.
+  const defaultActionButton = {
+    labelKey: 'continue',
+    route: getRouting(form),
+  }
+  const reviewActionButton = {
+    labelKey: 'updateAndReturn',
+    route: '/review',
+  }
+  // Set up routing to determine if the user is reviewing previously entered data.
+  const router = useRouter()
+  // If the user is reviewing previously entered data, use the review button.
+  // Otherwise, use the default button.
+  const [continueBtn, setContinueBtn] = useState(
+    router.query.mode === 'review' ? reviewActionButton : defaultActionButton
```

Comment:
@karinamzalez I wanted to pull this check (`router.query.mode === 'review'`) into a separate utils file but I was getting errors about trying to call `useRouter`, which is a hook, in a non-hook/non-component context. I could use some brainstorming on this with you.

Reply:
talked to @karinamzalez about this. Consider moving this to `_app.tsx` and pass it in as a prop